lonestar/eda/cpu/SPRoute/utility.h

```diff
@@ -1207,7 +1207,7 @@ void StNetOrder() {
 
       gridsX = treeedge->route.gridsX;
       gridsY = treeedge->route.gridsY;
-      for (i = 0; i <= treeedge->route.routelen; i++) {
+      for (i = 0; i < treeedge->route.routelen; i++) {

```

Comment:
@Michael-JY-He I'm not entirely certain this fix is right. I'd appreciate your input here.

Reply:
I don't remember I've change this line. Are you going to change it?

The line without "=" is correct. This loop tries to compute something between route point, which means it needs to iterate routelen - 1 times.

Reply:
Yes, the changes shown here are changes that I'm proposing we make in master. It'll be in master once I merge. Since this version is correct I'll go ahead and merge this. This will help with some (but not all) of the issues mentioned in https://github.com/IntelligentSoftwareSystems/Galois/issues/179. Thanks!

Reply:
Okay, these changes are visible in the current master now.

Reply:
Oh I was wrong. I'm Sorry.

The routelen is already -1.  The line with "=" is correct.

Reply:
Okay, feel free to change it back in your next round of edits. With the "=" there though, it has an out-of-bounds memory access that gets picked up by the sanitizer, so something is wrong in that case.

Reply:
Removing the "=" sign was my best guess at what the fix was without having more context on what the app was doing.